### PR TITLE
More useful error messages.

### PR DESCRIPTION
**Pull request by WrinklyNinja**
Applied the patch given in jbeder/yaml-cpp#200 with the correct code style (as far as I can see, I don't use clang, so can't run clang-format).

### CODE REVIEW

File: include/yaml-cpp/exceptions.h [hunk 1]
```diff
@@ -120,6 +120,9 @@ class Exception : public std::runtime_error {
  private:
   static const std::string build_what(const Mark& mark,
                                       const std::string& msg) {
+    if (mark.is_null())
```
Comment (jbeder) [inline include/yaml-cpp/exceptions.h]: Please add curly braces for this "if" statement, even if it's a one-line fix. I've been doing that with new code - I wish there was a way of telling clang-format to do this automatically. I may go through and update all of the one-line control flow statements at some point, but for now, I prefer new code to be in the proper format.

File: include/yaml-cpp/exceptions.h [hunk 2]
```diff
@@ -178,14 +181,14 @@ class InvalidNode : public RepresentationException {
 
 class BadConversion : public RepresentationException {
  public:
-  BadConversion()
-      : RepresentationException(Mark::null_mark(), ErrorMsg::BAD_CONVERSION) {}
+  BadConversion(const Mark& mark_)
```
Comment (jbeder) [inline include/yaml-cpp/exceptions.h]: explicit

File: include/yaml-cpp/node/detail/node.h [hunk 3]
```diff
@@ -64,6 +65,9 @@ class node : private boost::noncopyable {
     m_pRef->set_data(*rhs.m_pRef);
   }
 
+  void set_mark(const Mark& mark) {
+    m_pRef->set_mark(mark);
+  }
   void set_type(NodeType::value type) {
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/node.h]: add newline

File: include/yaml-cpp/node/impl.h [hunk 4]
```diff
@@ -66,6 +66,12 @@ inline bool Node::IsDefined() const {
   return m_pNode ? m_pNode->is_defined() : true;
 }
 
+inline Mark Node::Mark() const {
+  if (!m_isValid)
```
Comment (jbeder) [inline include/yaml-cpp/node/impl.h]: add braces

File: include/yaml-cpp/exceptions.h [hunk 5]
```diff
@@ -178,14 +181,14 @@ class InvalidNode : public RepresentationException {
...
+      : RepresentationException(mark_, ErrorMsg::BAD_CONVERSION) {}
 };
 
 template <typename T>
 class TypedBadConversion : public BadConversion {
  public:
-  TypedBadConversion() : BadConversion() {}
+  TypedBadConversion(const Mark& mark_) : BadConversion(mark_) {}
```
Comment (jbeder) [inline include/yaml-cpp/exceptions.h]: explicit

**Comment by jbeder** [comment]
Thanks for the patch. I made a few stylistic comments, but overall looks good.

(Note for the future: you can download clang-format as a standalone binary. Ideally, you'd do that and run it :) for now, though, I just want to confirm - you're using two spaces for indent, correct? Everything else looks correct format-wise, and I'm not confident in github's ability to tell me what's a tab and what's a space.)

**Comment by WrinklyNinja** [comment]
Do you want me to squash the code style commit into my previous commit, or leave them as-is?

**Comment by jbeder** [comment]
Please squash.

**Comment by WrinklyNinja** [comment]
Done.

**Comment by jbeder** [comment]
Thanks for the pull request. Man, I'm super disappointed in github's code review process. Oh well.

**Pull request merged**